src/query/storages/system/src/indexes_table.rs

```diff
@@ -0,0 +1,123 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use common_catalog::catalog::CATALOG_DEFAULT;
+use common_catalog::plan::PushDownInfo;
+use common_catalog::table::Table;
+use common_exception::Result;
+use common_expression::types::StringType;
+use common_expression::DataBlock;
+use common_expression::FromData;
+use common_expression::TableDataType;
+use common_expression::TableField;
+use common_expression::TableSchemaRefExt;
+use common_meta_app::schema::ListIndexesReq;
+use common_meta_app::schema::TableIdent;
+use common_meta_app::schema::TableInfo;
+use common_meta_app::schema::TableMeta;
+use common_storages_fuse::TableContext;
+
+use crate::table::AsyncOneBlockSystemTable;
+use crate::table::AsyncSystemTable;
+
+pub struct IndexesTable {
+    table_info: TableInfo,
+}
+
+#[async_trait::async_trait]
+impl AsyncSystemTable for IndexesTable {
+    const NAME: &'static str = "system.indexes";
+
+    fn get_table_info(&self) -> &TableInfo {
+        &self.table_info
+    }
+
+    async fn get_full_data(
+        &self,
+        ctx: Arc<dyn TableContext>,
+        _push_downs: Option<PushDownInfo>,
+    ) -> Result<DataBlock> {
+        let tenant = ctx.get_tenant();
+        let catalog = ctx.get_catalog(CATALOG_DEFAULT)?;
+        if let Some(indexes) = catalog
+            .list_indexes(ListIndexesReq {
+                tenant,
+                table_id: None,
+            })
+            .await?
+        {
+            let mut names = Vec::with_capacity(indexes.len());
+            let mut types = Vec::with_capacity(indexes.len());
+            let mut tables = Vec::with_capacity(indexes.len());
+            let mut defs = Vec::with_capacity(indexes.len());
+            let mut created_on = Vec::with_capacity(indexes.len());
+
+            for (_, index) in indexes {
+                names.push(index.ident.index_name.as_bytes().to_vec());
+                types.push(index.index_type.to_string().as_bytes().to_vec());
+                tables.push(index.table_desc.as_bytes().to_vec());
+                defs.push(index.query.as_bytes().to_vec());
+                created_on.push(
+                    index
+                        .created_on
+                        .format("%Y-%m-%d %H:%M:%S.%3f %z")
+                        .to_string()
+                        .as_bytes()
+                        .to_vec(),
+                );
+            }
+
+            Ok(DataBlock::new_from_columns(vec![
+                StringType::from_data(names),
+                StringType::from_data(types),
+                StringType::from_data(tables),
+                StringType::from_data(defs),
+                StringType::from_data(created_on),
```

Comment:
created_on  could use Timestamp DataType.

Reply:
In other system tables (like `system.tables`), it is string type.